website/blog/2018-07-27-removing-babels-stage-presets.md

```diff
@@ -0,0 +1,189 @@
+---
+layout: post
+title:  "Removing Babel's Stage Presets"
+author: Henry Zhu
+authorURL: https://twitter.com/left_pad
+date:   2018-07-27 12:00:00
+categories: announcements
+share_text: "Removing Babel's Stage Presets"
+---
+
+Moving forward with v7, we've decided it's best to stop publishing the Stage presets in Babel (e.g. `@babel/preset-stage-0`).
+
+We didn't make this decision lightly and wanted bring show some context regarding TC39, new proposals, and Babel.
+
+<!--truncate-->
+
+## Some History
+
+A Babel preset is a shareable list of plugins.
+
+The [official Babel Stage presets](https://babeljs.io/docs/en/next/presets) tracked the [TC39 Staging process](https://tc39.github.io/process-document/) for new [syntax proposals](https://github.com/tc39/proposals) in JavaScript.
+
+Each preset (ex. `stage-3`, `stage-2`, etc.) included all the plugins for that particular stage and the ones above it. For example, `stage-2` included `stage-3`, and so on.
+
+---
+
+This allowed users who wanted to use experimental syntax to simply add the preset, instead of needing to configure/install each individual plugin.
+
+We actually [added](https://github.com/babel/babel/pull/2649) the Stage presets shortly after Babel's v6 release (it was previously a config flag in v5).
+
+These are some older examples from Babel v6.
+
+It’s common to see this in a config: 
+
+```js
+{
+  "presets": ["es2015", "react", "stage-0"]
+}
+```
+
+Original source of the package: [babel-preset-stage-0](https://unpkg.com/babel-preset-stage-0@6.0.14/index.js)
+
+```js
+module.exports = {
+  presets: [
+    require("babel-preset-stage-1")
+  ],
+  plugins: [
+    require("babel-plugin-transform-do-expressions"),
+    require("babel-plugin-transform-function-bind")
+  ]
+};
+```
+
+## Problems
+
+These presets were a convenient way to use what we all wanted: the new, shiny, "yet-to-be-determined" future of JavaScript.
+
+Looking back, it worked really well! (Maybe too well?)
+
+### Too Good a Job?
+
+Languages like [CoffeeScript](https://coffeescript.org/) and tooling like [Traceur](https://github.com/google/traceur-compiler) helped establish the idea of compiling JavaScript. Babel made it even easier to both use new/future syntax and integrate with existing tooling. The expectations changed from skepticism and worry to completely embracing the experimental.
+
+We probably wouldn't be where we are at if not for the wide adoption of compilers such as Babel: it accelerated the usage (and teaching) of ES2015 to a much larger audience. The growth of React further fueled usage as its JSX syntax, class properties, and object rest/spread led to people knowing a bit more about these syntax proposals.
```

Comment:
Minor nit, but "where we are at" reads a bit awkwardly to me. I'd suggest changing it to one of the following, which I think are both more common:

- "where we're at"
- "where we are" (remove "at")